akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala

```diff
@@ -12,12 +12,40 @@ import StatusCodes.ClientError
  * repeating anything present in the message itself (in order to not open holes for XSS attacks),
  * while the detail can contain additional information from any source (even the request itself).
  */
-final case class ErrorInfo(summary: String = "", detail: String = "") {
+final class ErrorInfo(
+  val summary:         String = "",
+  val detail:          String = "",
+  val errorHeaderName: String = ""
+) extends scala.Product with scala.Serializable with scala.Equals with java.io.Serializable {
   def withSummary(newSummary: String) = copy(summary = newSummary)
   def withSummaryPrepended(prefix: String) = withSummary(if (summary.isEmpty) prefix else prefix + ": " + summary)
+  def withHeaderNamePrepend(headerName: String) = setErrorHeaderName(errorHeaderName = headerName)
   def withFallbackSummary(fallbackSummary: String) = if (summary.isEmpty) withSummary(fallbackSummary) else this
   def formatPretty = if (summary.isEmpty) detail else if (detail.isEmpty) summary else summary + ": " + detail
   def format(withDetail: Boolean): String = if (withDetail) formatPretty else summary
+
+  private[akka] def copy(summary: String = summary, detail: String = detail): ErrorInfo = {
+    new ErrorInfo(summary, detail, errorHeaderName)
+  }
+
+  private[akka] def setErrorHeaderName(errorHeaderName: String): ErrorInfo = new ErrorInfo(summary, detail, errorHeaderName)
```

Comment:
Perhaps make this public and call it `withErrorHeaderName`?

Reply:
Surely you are right. delete this method and rename `withHeaderNamePrepend` method to `withErrorHeaderName`.